### ISSUE TITLE
[rush] use globalOverrides in `pnpm-config.json` always show Dependencies of project do not match the current shrinkwrap.

### ISSUE THREAD

**Issue by CupNoodleFork** [microsoft/rushstack#3978]
## Summary
When I configure `globalOverrides` in `pnpm-config.json`
like this
```
"globalOverrides": {
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-is": "17.0.2"
  }
```
Then there will always be a warning everytime when i run `rush update`
![image](https://user-images.githubusercontent.com/17742391/219542793-cb900c66-2b85-4096-b03f-1354c5507cc6.png)

## Repro steps

Here is a minimal reproducible project
https://github.com/CupNoodleFork/rush-pnpm-config-shrinkwrap-mismatch

  **Expected result:** 

  **Actual result:** 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.90.2 |
| `rushVersion` from rush.json? | 5.90.2 |
| `useWorkspaces` from rush.json? | true |
| Operating system? | mac |
| Would you consider contributing a PR? | NO |
| Node.js version (`node -v`)? | v14.19.1 |

**Comment by dmichon-msft** [comment]
Why are you using `pnpm-config.json` to override the direct declared dependencies of your own local workspace packages? The expectation of the tooling is that you don't need to use install-time overrides to correct your own declared dependencies, only dependencies of dependencies.

**Comment by CupNoodleFork** [comment]
@dmichon-msft You are right. Most of the time, however, i cannot clearly distinguish between dependencies and dependencies of dependencies. It is possible that a module will exist in deps or deps of deps.
In my project, the react version in dependencies is indeterminate, it will lead the "more than one copy of React in the same app" problem.

**Comment by dmichon-msft** [comment]
The warning will only happen if the installed version (from the overrides) does not satisfy the specifier (range) in your local package.json files. It's perfectly fine to use globalOverrides to constrain said version to an exact version.

i.e. if you want to support both 17 and 18, but install 17, put `"react": "^17 || ^18"` (or similar) in your package.json, and then globally override it to `"react": "17.0.2"`, etc.

**Comment by CupNoodleFork** [comment]
In out project, It might be correct to use the `ensureConsistentVersions` feature to constrain the dependency's version to an exact one via `preferredVersions` and `allowedAlternativeVersions`. However, We have dozens of projects, it needs a huge workload.
